src/azfilenfs-watchdog

```diff
@@ -0,0 +1,222 @@
+#
+while :; do
+    sleep $MONITOR_INTERVAL_SECS
+
+    epoch_now=$(date +%s)
+
+    #
+    # Go over all lines in $AZ_FILES_MOUNTMAP and check them for two things:
+    # 1. Is that entry still in use by at least one aznfs mount, if not remove the entry.
+    # 2. Is stunnel process running?
+    #
+    # We store the mtime of AZ_FILES_MOUNTMAP while inside the lock so that if any mount helper process
+    # updates it after this we will skip modification for sake of safety. We will come to it
+    # in the next iteration when it's safer.
+    #
+    exec {fd}<$AZ_FILES_MOUNTMAP
+    flock -e $fd
+    mtime_mountmap=$(stat -c%Y $AZ_FILES_MOUNTMAP)
+    IFS=$'\n' lines=$(cat $AZ_FILES_MOUNTMAP)
+    flock -u $fd
+    exec {fd}<&-
+
+    #
+    # findmnt must be done after reading AZ_FILES_MOUNTMAP so that if we come accross a
+    # AZ_FILES_MOUNTMAP entry whose all nfs file shares are unmounted, we know
+    # for sure that it's not in use by any mount and can be removed.
+    #
+    findmnt=$(findmnt | grep 'nfs4\|127.0.0.1' 2>&1)
+
+    #
+    # For no matching mounts also, findmnt exits with a failure return, so check
+    # for both exit status and non-empty error o/p.
+    #
+    if [ $? -ne 0 -a -n "$findmnt" ]; then
+        eecho "${findmnt}."
+        eecho "[FATAL] findmnt failed unexpectedly!"
+        eecho "[FATAL] Azfilenfs-watchdog service is exiting, will not monitor Azure NFS file shares."
+        eecho "[FATAL] Please contact Microsoft support before using any NFS File shares."
+        # This usually indicates some non-transient issue, bail out.
+        exit 1
+    fi
+
+    for line in $lines; do
+        if [ -z "$line" ]; then
+            continue
+        fi
+
+        #
+        # AZ_FILES_MOUNTMAP line is of the form:
+        # account.file.preprod.core.windows.net;<stunnel_account.file.preprod.core.windows.net.conf path>;<stunnel_account.file.preprod.core.windows.net.log path>;<stunnel_account.file.preprod.core.windows.net.pid path>;<checksumHash>
+        #
+        IFS=";" read l_host l_conf l_log l_pid l_checksumhash <<< "$line"
+
+        if [ -z "$l_host" -o -z "$l_conf" -o -z "$l_pid" ]; then
+            wecho "[FATAL] Deleting invalid line in $AZ_FILES_MOUNTMAP: [$line]!"
+            l_mtime=$(ensure_az_files_mountmap_not_exist "$line")
+            [ $? -eq 0 ] && mtime_mountmap=$l_mtime
+            continue
+        fi
+
+	accept_port=$(cat $l_conf | grep accept | cut -d ':' -f 2)
+	echo "accept_port: $accept_port"
+
+        #
+        # Delete entry from AZ_FILES_MOUNTMAP if there are no mounted shares on that host.
+        # As long as we have at least one mount using the AZ_FILES_MOUNTMAP entry, we leave
+        # it around.
+        #
+        if ! echo "$findmnt" | grep "$accept_port" >/dev/null; then
+                pecho "No mounted shares for host $l_host, deleting from ${AZ_FILES_MOUNTMAP} [$line]."
+
+		# Kill stunnel process first.
+		pid=$(cat $l_pid)
+		kill -9 $pid
```

Comment:
script may need privileges to run the findmnt and kill command. how to make sure the scripts have the right privileges to run

Reply:
We run watchdog script with root privileges, so running kill command wouldn't be a problem. Plus, findmnt can be run without root privileges too.